src/main/java/org/opentripplanner/transit/raptor/rangeraptor/ExactTripSearchState.java

```diff
@@ -0,0 +1,40 @@
+package org.opentripplanner.transit.raptor.rangeraptor;
+
+import java.util.Arrays;
+import org.opentripplanner.transit.raptor.rangeraptor.transit.TripScheduleSearchWindow;
+
+/**
+ * Stores the previous iteration's access times, to allow limiting boardings to those found since
+ * the previous iteration.
+ *
+ * @see org.opentripplanner.transit.raptor.rangeraptor.transit.TripScheduleExactMatchSearch
+ */
+public class ExactTripSearchState implements TripScheduleSearchWindow {
+
+    private final int numberOfStops;
+    private int[] previousAccessTimes;
+    private int[] currentAccessTimes;
```

Comment:
Keeping the memory usage as low as possible is key to avoid fetching data outside the second level cashe, so even if this adds just 8 bytes for each stop it is significant. Also, I think it is unnecessary complicated and pollute the exact trip search. Just disabling it for when a time-restricted-access is added seams much easier.